apis/v1beta1/config.go

```diff
@@ -75,6 +87,37 @@ func (c *AnyConfig) MarshalJSON() ([]byte, error) {
 	return json.Marshal(c.Object)
 }
 
+// Pipeline is a struct of component type to a list of component IDs.
+type Pipeline struct {
+	Receivers  []string `json:"receivers" yaml:"receivers"`
+	Processors []string `json:"processors" yaml:"processors"`
+	Exporters  []string `json:"exporters" yaml:"exporters"`
+}
+
+// GetEnabledComponents constructs a list of enabled components by component type.
+func (c *Config) GetEnabledComponents() map[ComponentType]map[string]interface{} {
```

Comment:
eventually this will replace the way we do this today in config_validate.go but we can't do that for now...

Reply:
How are the connectors handled?

Reply:
connectors don't have their own definition in a pipeline, they are defined in the config and [used as exporters and receivers](https://github.com/open-telemetry/opentelemetry-collector/tree/main/connector#usage), this will work as expected for them as a result :)